front/components/vaults/VaultAppsList.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Chip,
+  CommandLineIcon,
+  DataTable,
+  Searchbar,
+  Spinner,
+} from "@dust-tt/sparkle";
+import type {
+  ConnectorType,
+  DataSourceViewCategory,
+  PlanType,
+  VaultType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import type { CellContext } from "@tanstack/react-table";
+import type { ComponentType } from "react";
+import { useRef } from "react";
+import { useState } from "react";
+import * as React from "react";
+
+import { useApps } from "@app/lib/swr";
+import { classNames } from "@app/lib/utils";
+
+type RowData = {
+  category: string;
+  label: string;
+  icon: ComponentType;
+  connector?: ConnectorType;
+  fetchConnectorError?: string;
+  visibility: string;
+  workspaceId: string;
+  onClick?: () => void;
+};
+
+type Info = CellContext<RowData, unknown>;
+
+type VaultResourcesListProps = {
+  owner: WorkspaceType;
+  plan: PlanType;
+  isAdmin: boolean;
+  vault: VaultType;
+  systemVault: VaultType;
+  category: DataSourceViewCategory;
+  onSelect: (sId: string) => void;
+};
+
+const getTableColumns = () => {
+  return [
+    {
+      header: "Name",
+      accessorKey: "label",
+      id: "label",
+      cell: (info: Info) => (
```

Comment:
instead of defining an `Info` type wit unknown cell value, use `CellContext<RowData, string>` here with the real type associated to the column, to leverage tanstack table. 
Also rather use use `info.getValue()` instead of `info.row.original.label` to get the value associated with the row .